mongodbatlas/fw_resource_mongodbatlas_alert_configuration.go

```diff
@@ -0,0 +1,811 @@
+	}
+
+	if !reflect.DeepEqual(alertConfigPlan.Matcher, alertConfigState.Matcher) {
+		apiReq.Matchers = expandTFAlertConfigurationMatchers(alertConfigPlan.Matcher)
+	}
+
+	// Always refresh structure to handle service keys being obfuscated coming back from read API call
+	notifications, err := expandTFAlertConfigurationNotification(alertConfigPlan.Notification)
+	if err != nil {
+		resp.Diagnostics.AddError(errorUpdateAlertConf, err.Error())
+		return
+	}
+	apiReq.Notifications = notifications
+
+	var updatedAlertConfigResp *matlas.AlertConfiguration
+
+	// Cannot enable/disable ONLY via update (if only send enable as changed field server returns a 500 error) so have to use different method to change enabled.
+	if reflect.DeepEqual(apiReq, &matlas.AlertConfiguration{Enabled: pointy.Bool(true)}) ||
+		reflect.DeepEqual(apiReq, &matlas.AlertConfiguration{Enabled: pointy.Bool(false)}) {
+		// this code seems unreachable, as notifications are always being set
+		updatedAlertConfigResp, _, err = conn.AlertConfigurations.EnableAnAlertConfig(ctx, ids[encodedIDKeyProjectID], ids[encodedIDKeyAlertID], apiReq.Enabled)
+	} else {
+		updatedAlertConfigResp, _, err = conn.AlertConfigurations.Update(ctx, ids[encodedIDKeyProjectID], ids[encodedIDKeyAlertID], apiReq)
+	}
+
+	if err != nil {
+		resp.Diagnostics.AddError(errorUpdateAlertConf, err.Error())
+		return
+	}
+
+	newAlertConfigurationState := toTFAlertConfigurationModel(updatedAlertConfigResp, &alertConfigPlan)
+
+	// save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &newAlertConfigurationState)...)
+}
+
+func (r *AlertConfigurationRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	conn := r.client.Atlas
+
+	var alertConfigState tfAlertConfigurationRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &alertConfigState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ids := decodeStateID(alertConfigState.ID.ValueString())
+
+	_, err := conn.AlertConfigurations.Delete(ctx, ids[encodedIDKeyProjectID], ids[encodedIDKeyAlertID])
+	if err != nil {
+		resp.Diagnostics.AddError(errorReadAlertConf, err.Error())
+	}
+}
+
+func (r *AlertConfigurationRS) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.SplitN(req.ID, "-", 2)
+
+	if len(parts) != 2 {
+		resp.Diagnostics.AddError("import format error: to import a cluster, use the format {project_id}-{id}", "")
+	}
+
+	projectID := parts[0]
+	alertConfigurationID := parts[1]
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), encodeStateID(map[string]string{
+		"id":         alertConfigurationID,
+		"project_id": projectID,
+	}))...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectID)...)
+}
+
+func expandTFAlertConfigurationNotification(tfNotificationSlice []tfNotificationModel) ([]matlas.Notification, error) {
```

Comment:
same as above. I would like to move away from expand - flat format and use `new[NameOfResource]` which is widely used in golang

Reply:
+1

Reply:
agree to moving forward with this convention. Just to be clear, for the case of converting from api models to terraform models I see that the toTF* prefix is being used (in project resource and my changes). Are we in agreement to rename these to newTF*  as well?

Reply:
my vote is for `newTF*` 👍

Reply:
adjusted.